### PR TITLE
JAMES-3302 Keep improving Distributed Server documentation

### PR DESCRIPTION
**Pull request by chibenwa**
- Architecture section
 - Include some images
 - Reorganise (and refresh) webadmin section
 - Metrics

### CODE REVIEW

File: docs/modules/servers/pages/distributed/configure/remote-delivery-error-handling.adoc [hunk 1]
```diff
@@ -0,0 +1,107 @@
+= About RemoteDelivery error handling
+
+The advanced server mailQueue implemented by combining RabbitMQ for messaging and Cassandra for administrative operation
+do not support delays.
```
Comment (rouazana) [inline docs/modules/servers/pages/distributed/configure/remote-delivery-error-handling.adoc:4]:
```suggestion
does not support delays.
```

File: docs/modules/servers/pages/distributed/configure/remote-delivery-error-handling.adoc [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+Delays are an important feature for Mail Exchange servers, allowing to defer in time the retries, potentially letting the
+time for the remote server to recover. Furthermore, they enable implementation of advanced features like throttling and
+rate limiting of emails sent to a given domain.
+
+As such, the use of the distributed server as a Mail Exchange server is currently discouraged.
+
+However, for operators willing to inter-operate with a limited set of well-identified, trusted remote mail servers, such
```
Comment (rouazana) [inline docs/modules/servers/pages/distributed/configure/remote-delivery-error-handling.adoc:12]: you also have to explain the limit of this solution, in particular that repositories are not meant to store transient data and so could be prone to tombstones.

File: docs/modules/servers/pages/distributed/configure/remote-delivery-error-handling.adoc [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            <mailet match="All" class="ToRepository">
+                <repositoryPath>cassandra://var/mail/error/</repositoryPath>
+            </mailet>
+        </processor>
+....
+
+Note:
+
+- The *relay* processor holds a RemoteDelivery mailet configured to do a single try, at most 5 times (see the AtMost matcher).
+Mails exceeding the AtMost condition are concidered as permanent delivery errors. Delivery errors are sent to the
```
Comment (rouazana) [inline docs/modules/servers/pages/distributed/configure/remote-delivery-error-handling.adoc:80]:
```suggestion
Mails exceeding the AtMost condition are considered as permanent delivery errors. Delivery errors are sent to the
```

File: docs/modules/servers/pages/distributed/configure/remote-delivery-error-handling.adoc [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+*remote-delivery-error* processor.
+- The *remote-delivery-error* stores temporary and permanent errors.
+- Permanent relay errors are stored in `cassandra://var/mail/error/remote-delivery/permanent/`.
+- Temporary relay errors are stored in `cassandra://var/mail/error/remote-delivery/temporary/`.
+
+In order to retry the relay of temporary failed emails, operators will have to configured a cron job for reprocessing
```
Comment (rouazana) [inline docs/modules/servers/pages/distributed/configure/remote-delivery-error-handling.adoc:86]:
```suggestion
In order to retry the relay of temporary failed emails, operators will have to configure a cron job for reprocessing
```

File: docs/modules/servers/pages/distributed/configure/remote-delivery-error-handling.adoc (hunk 3 above)
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/configure/remote-delivery-error-handling.adoc]:
```suggestion
Mails exceeding the AtMost condition are considered as permanent delivery errors. Delivery errors are sent to the
```

File: docs/modules/servers/pages/distributed/configure/remote-delivery-error-handling.adoc (hunk 4 above)
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/configure/remote-delivery-error-handling.adoc]:
```suggestion
In order to retry the relay of temporary failed emails, operators will have to configure a cron job for reprocessing
```

File: docs/modules/servers/pages/distributed/configure/remote-delivery-error-handling.adoc [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+This can be achieved via the following webAdmin call :
+
+....
+curl -XPATCH 'http://ip:8000/mailRepositories/cassandra%3A%2F%2Fvar%2Fmail%2Ferror%2Fremote-delivery%2Ftemporary%2F/mails?action=reprocess&processor=relay'
+....
+
+See xref:distributed/operate/webadmin.adoc#_reprocessing_mails_from_a_mail_repository[the documentation].
+
+Administrator need to keep a close eye on permanent errors (that might require audit, and potentially contacting the remote
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/configure/remote-delivery-error-handling.adoc:106]:
```suggestion
Administrators need to keep a close eye on permanent errors (that might require audit, and potentially contacting the remote
```

File: docs/modules/servers/pages/distributed/configure/vault.adoc [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+Messages stored in the Deleted Messages Vault could be deleted after exceeding their retentionPeriod (explained below).
+It also supports to restore or export messages matching with defined criteria in
+xref:distributed/operate/webadmin.adoc#_deleted_messages_vault[WebAdmin deleted messages vault document] by using
+xref:distributed/operate/webadmin.adoc#_deleted_messages_vault[WebAdmin endpoints].
+
+== Deleted Messages Vault Configuration
+
+To make James use DeletedMessageVault, you need to configure the PreDeletionHook for it.
+
+How to do this is doculented in the xref:distributed/configure/listeners.adoc[*listeners.xml*]
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/configure/vault.adoc:13]:
```suggestion
How to do this is documented in the xref:distributed/configure/listeners.adoc[*listeners.xml*]
```

File: docs/modules/servers/pages/distributed/configure/webadmin.adoc [hunk 7]
```diff
@@ -0,0 +1,60 @@
+= webadmin.properties
+
+The web administration supports for now the CRUD operations on the domains, the users, their mailboxes and their quotas,
+managing mail repositories, performing cassandra migrations, and much more, as described in the following sections.
+
+*WARNING*: This API allow authentication only via the use of JWT. If not
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/configure/webadmin.adoc:6]:
```suggestion
*WARNING*: This API allows authentication only via the use of JWT. If not
```

File: docs/modules/servers/pages/distributed/extend/index.adoc [hunk 8]
```diff
@@ -16,3 +16,12 @@ add features to your SMTP server.
 
 Writing xref:distributed/extend/webadmin-routes.adoc[WebAdmin routes] enables to
 add features to the WebAdmin REST API.
+
+== Handling injections for your extensions
+
+(TODO)
+
+ - Using injections defined by the Distributing Server
+ - Adding custom injections injections to the one of the Distributed Server for injections in your extensions.
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/extend/index.adoc:25]:
```suggestion
 - Adding custom injections to one of the Distributed Server for injections in your extensions.
```

File: docs/modules/servers/pages/distributed/operate/webadmin.adoc [hunk 9]
```diff
@@ -143,6 +143,162 @@ Response codes:
 
 * 200: List of available health checks
 
+== Task management
+
+Some webadmin features schedules tasks. The task management API allow to
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/operate/webadmin.adoc:148]:
```suggestion
Some webadmin features schedule tasks. The task management API allow to
```

File: docs/modules/servers/pages/distributed/operate/webadmin.adoc [hunk 10]
```diff
@@ -143,6 +143,162 @@ Response codes:
...
+Response codes:
+
+* 200: The specific task was found and the execution report exposed
+above is returned
+* 400: Invalid task ID
+* 404: Task ID was not found
+
+=== Awaiting a task
+
+One can await the end of a task, then receive it’s final execution
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/operate/webadmin.adoc:200]:
```suggestion
One can await the end of a task, then receive its final execution
```

**Comment by chibenwa** [comment]
```
% grep TODO pages/distributed -R           
     pages/distributed/configure/blobstore.adoc:(TODO) https://github.com/apache/james-project/blob/master/src/site/xdoc/server/config-blobstore.xml 
    pages/distributed/configure/matchers.adoc:(TODO)
```

Almost there ;-)

We soon have a first iteration of the Distributed server documentation.

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**